src/main/routes/metrics-route.ts

```diff
@@ -36,20 +34,25 @@ async function loadMetrics(promQueries: string[], cluster: Cluster, prometheusPa
 
     return loaders.get(query) ?? loaders.set(query, loadMetricHelper()).get(query)
   }
-
   return Promise.all(queries.map(loadMetric))
 }
 
 class MetricsRoute extends LensApi {
   async routeMetrics({ response, cluster, payload, query }: LensApiRequest) {
     const queryParams: IMetricsQuery = Object.fromEntries(query.entries())
-
+    const prometheusMetadata: ClusterPrometheusMetadata = {}
     try {
       const [prometheusPath, prometheusProvider] = await Promise.all([
         cluster.contextHandler.getPrometheusPath(),
         cluster.contextHandler.getPrometheusProvider()
       ])
-
+      prometheusMetadata["provider"] = prometheusProvider?.id
+      prometheusMetadata["autoDetected"] = !cluster.preferences.prometheusProvider?.type
+      if (!prometheusPath) {
+        prometheusMetadata["success"] = false
```

Comment:
Why access  to `prometheusMetadata` via `["fieldName"]`? Some intention?

Reply:
just leftovers when I did not have proper type for prometheusMetadata. Now it's there so I will change those. Thanks for noticing!